Automattic/jetpack
# Search: support search through multiple blogs

Fixes #26031 

#### Changes proposed in this Pull Request:
The PR adds passing `additional_blog_ids` to the the search endpoint to support search through multiple blogs. The API support was added in `D74174-code`. And recently requested by @pablohoneyhoney in `p5uIfZ-cqA#comment-17850-p2`. The support is only for Instant Search.

Known issue: blog icon and blog name might be empty.

#### Other information:


#### Jetpack product discussion
`p5uIfZ-cqA#comment-17850-p2`
`panfyZ-pf-p2`

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
* Prepare a main blog A and an additional blog B, both have Instant Search turned on
* Go to your sandbox, allow search blog B from blog A with wpsh
* `switch_to_blog(A);`
* `update_option('jetpack_allowed_xsite_search_ids', [B]);` <- Ensure B is an integer
* Add the following snippet to blog A
```PHP
add_filter( 'jetpack_instant_search_options', function( $options ) {
	$options['additionalBlogIds'] = array( 189825737 );
	return $options;
} );
```
* Search on blog A
* Ensure results from blog B comes up too
